File: python/visualization.py
# encoding=utf8

import matplotlib, matplotlib.pyplot as plt, numpy as np

matplotlib.style.use('ggplot')

def plot_results(expected, received):
    """
    Displays the result of the computed model.

    :param expected the actual value of the rental price
    :param received the value estimated by the model
    """
    plt.figure()
    plt.subplot(211)
    plt.plot(expected, received, 'ro')
    plt.xlabel(u"Preço Real")
    plt.ylabel(u"Preço Estimado")
    plt.title(u"Resultados para set de teste")

    plt.subplot(212)
    plt.hist(np.abs(expected - received), bins=30)
    plt.xlabel(u"Número de casos")
    plt.ylabel(u"Erro absoluto")

    plt.xlabel(u"Erro")
    plt.ylabel(u"Quantidade")
    plt.title("Resultados para o para teste")

def plot_scater_map(data):
    """
    Plots a map and the location of each rental unit at their coordinates.
    The colormap represents the rental price. Size of the points is inversely
    proportional to the price.

    :param data: the rental units tho be shown in the map
    """
    plt.figure();
    im = matplotlib.image.imread('flnp2.png');
    im[:, :, -1] = 0.7;
    plt.imshow(im, extent=[-48.68, -48.3, -27.85, -27.35])
    plt.scatter(data.longitude,
                data.latitude,
                c=data.rentPrice,
                s = 2000/np.sqrt(data.rentPrice))

    plt.colorbar()
    plt.axis([-48.68, -48.3, -27.85, -27.35])
    plt.title(u"Distribução dos Imóveis")

File: R/visualization.R

library(ggplot2)
library(RMisc)
library(grid)
library(png)


plot_results <- function(expected, predicted) {
  
  par(mfrow=c(2,1)) 
  
  plot(test_data$rentPrice ~ predicted, 
       xlab = "Valor real (R$)", 
       ylab = "Preço estimado (R$)", 
       col = 'red',
       main = "Relação entre resultados obtidos")
  
  hist(abs(test_data$rentPrice - predicted), 
       breaks = 30, 
       c='red', 
       xlab = "Erro Absoluto (R$)", 
       ylab = "Número de casos", 
       main="Erro de medição")  
}

plot_results_ggplot2 <- function(expected, predicted) {
  
  p1 <- ggplot(data.frame(expected,predicted), aes(x=expected, y=predicted)) +
    xlab("Preço do aluguel") +
    ylab("Preço Estimado") +
    geom_point(colour='red') +
    ggtitle("Relação entre resultados obtidos")
  
  p2 <- ggplot(abs(expected- predicted), aes(x=expected) ) + 
    geom_histogram(fill='red', colour='grey') +
    xlab("Erro absoluto") +
    ylab(" ")
  multiplot(p1, p2, cols=1)
}

plot_map <- function(values) {
  img <- readPNG("flnp2.png")
  img[,,4] <- 0.3
  ggplot(train_data, aes(x=longitude, y=latitude, color = rentPrice) ) +
    scale_colour_gradientn( 
      colours=rev(rainbow(4)), guide = "colourbar") +
    annotation_custom(
      rasterGrob(img, width=unit(1,"npc"), height=unit(1,"npc")), 
      -Inf, Inf, -Inf, Inf) +
    scale_x_continuous(limits = c(-48.68, -48.3))+
    scale_y_continuous(limits = c(-27.85, -27.35))+
    scale_radius(range=c(1,6)) +
    geom_point(aes(size = 1000/ sqrt(train_data$rentPrice)))+
    ggtitle("Distribuição dos Imóveis")
} 